modules/andes-core/broker/src/main/java/org/dna/mqtt/moquette/messaging/spi/impl/ProtocolProcessor.java

```diff
@@ -833,6 +833,9 @@ void processSubscribe(ServerChannel session, SubscribeMessage msg, String client
         }
 
         MQTTAuthorizationSubject authSubject = authSubjects.get(clientID);
+        if (authSubject == null) {
```

Comment:
Can you please explain the reason for this Null check? 
Should we send a failure ack in this scenario?